client/asset/interface.go

```diff
@@ -326,7 +330,7 @@ type Wallet interface {
 	// NOTE: This could potentially be a long and expensive operation if
 	// performed long after the swap is broadcast; might be better executed from
 	// a goroutine.
-	FindRedemption(ctx context.Context, coinID, contract dex.Bytes) (redemptionCoin, secret dex.Bytes, err error)
+	FindRedemption(ctx context.Context, coinID, contractData dex.Bytes, deets *dex.SwapContractDetails) (redemptionCoin, secret dex.Bytes, err error)
```

Comment:
I was on the fence about adding these arguments, because I think we could probably work it out without them, through the tx cache or something. But all of the info in the `SwapContractDetails` is required to be saved by the client already, so it actually seems more on-model to have `Core` supply it. Open to suggestions though.